source/io/module_javaScriptInvoke.js

```diff
@@ -74,32 +74,121 @@ var assignJavaScriptInvoke;
             return returnValueRef;
         };
 
-        var cookieToJsValueMap = new Map();
-        var jsValueToCookieCache = new Map();
-        var jsRefNumCookieCounter = 0;
+        var generateUniqueRefNumCookie = (function () {
+            var jsRefNumCookieCounter = 0;
+            return function () {
+                jsRefNumCookieCounter += 1;
+                return jsRefNumCookieCounter;
+            };
+        }());
 
-        var cacheRefNum = function (cookie, jsValue) {
-            cookieToJsValueMap.set(cookie, jsValue);
-            jsValueToCookieCache.set(jsValue, cookie);
-        };
+        class StaticRefnumManager {
+            constructor () {
+                this._cookieToJsValueMap = new Map();
+                this._jsValueToCookieCache = new Map();
+            }
 
-        var hasCachedRefNum = function (cookie) {
-            var refNumExists = cookieToJsValueMap.has(cookie);
-            return refNumExists;
-        };
+            set (jsValue) {
```

Comment:
I find this name a little bit odd. A "set" that returns a value. How about getCookie instead of set and getValue for get?

Reply:
I went with createCookie and lookupValue thanks for the suggestions!